rundeckapp/grails-spa/packages/ui-trellis/src/app/components/activity/tests/activityList.spec.ts

```diff
@@ -0,0 +1,363 @@
+import {
+  rundeckServiceMock,
+  rundeckClientMock,
+  axiosMock,
+  i18nMocks,
+  mockEventBus,
+  mockReports,
+} from "../mocks/mock";
+import { flushPromises, shallowMount, VueWrapper } from "@vue/test-utils";
+import ActivityList from "../activityList.vue";
+import ActivityFilter from "../activityFilter.vue";
+import OffsetPagination from "../../../../library/components/utils/OffsetPagination.vue";
+import { cloneDeep } from "lodash";
+import { Modal, Btn } from "uiv";
+jest.mock("../../../../library/rundeckService", () => rundeckServiceMock);
+jest.mock("@rundeck/client", () => rundeckClientMock);
+jest.mock("axios", () => axiosMock);
+const mockRunningExecutions = mockReports.map((report) => report.execution);
+type ActivityListInstance = InstanceType<typeof ActivityList>;
+const shallowMountActivityList = async (
+  props = {},
+  options: Record<string, any> = {}, // Use a more generic type for options
+) => {
+  const wrapper = shallowMount(ActivityList, {
+    props: {
+      eventBus: mockEventBus,
+      displayMode: "mode",
+      ...props,
+    },
+    global: {
+      components: {
+        OffsetPagination,
+        ActivityFilter,
+        Modal,
+        Btn,
+      },
+      stubs: {
+        modal: false,
+        "i18n-t": true,
+        btn: false,
+        ProgressBar: true,
+      },
+      mocks: {
+        ...i18nMocks,
+        ...options.mocks, // Use the generic options object
+      },
+      directives: {
+        tooltip: () => {},
+      },
+      ...options, // Spread the generic options object
+    },
+  });
+  await wrapper.vm.$nextTick();
+  return wrapper as VueWrapper<ActivityListInstance>;
+};
+let reports;
+describe("ActivityList", () => {
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+  afterAll(() => {
+    jest.restoreAllMocks();
+    jest.useRealTimers();
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  beforeEach(() => {
+    reports = cloneDeep(mockReports);
+    axiosMock.get.mockImplementation((url) => {
+      if (url.includes("running")) {
+        return {
+          data: {
+            executions: [...mockRunningExecutions],
+          },
+        };
+      }
+      if (url.includes("eventsAjax")) {
+        return {
+          data: {
+            total: 20,
+            offset: 0,
+            reports: reports,
+          },
+        };
+      }
+    });
+  });
+
+  describe("renders", () => {
+    it("information about the page being shown and total number of results", async () => {
+      axiosMock.get.mockResolvedValue({
+        data: {
+          total: 20,
+          offset: 0,
+          reports: reports,
+        },
+      });
```

Comment:
As there's an axiosMock in the beforeEach hook, please remove all the mocks that are redundant (I know this one was added by me during our pairing, but now it can be removed).